docs/publishing/index.md

```diff
@@ -0,0 +1,125 @@
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when script changes state outside of the scope in which it is run. This includes when code in an es module alters values available on the `window` or `document`. A side effect occurs when registering Custom Elements via `customElements.define()`. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` takes care of this for you by creating a separate `my-element.js` that manages the registration of your custom element.
+
+## Do keep side effects out of your index.js
+
+When _not_ able to keep code with side effects out of your entry point, it should instead be put in a separate function that has to explicitly be called by the consumer (e.g. `run()`, `init()`, etc.). This prevents those side effects from occurring at run time.
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure as you do. To prevent dependency pointing from causing issue across different environments be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers
+
+When importing something that doesn't point directly to the entry point of a module (e.g.: `import {LitElement} from 'lit-element';`) always make sure to include a file extension (e.g. `import { ifDefined } from 'lit-html/directives/if-defined.js';`).
+
+## Optional: use `type: "module"`
+
+Using `type: "module"` in your `package.json` is used to help CDNs and bundlers choose whether to parse your package with module parse goal. However, it will also enforce strict realities on how es modules and common JS modules must be imported into your scripting. If all of the JS in your package is es modules, and using this entry in your `package.json` does not cause issues with your scripting, then this is a great way to support tooling the references your package in leveraging it more efficiently. If scripting in your package has issues running in Node@12+ due to this inclusion, excluding it will ease the local requirements on import statements in your code.
```

Comment:
```suggestion
Using `type: "module"` in your `package.json` is used to help CDNs and bundlers choose whether to parse your package as an actual module. However, it will also enforce strict realities on how es modules and common JS modules must be imported into your scripting or other tooling. If all of the JS in your package is es modules, and using this entry in your `package.json` does not cause issues with your scripting or tooling, then this is a great way to support tooling the references your package in leveraging it more efficiently. If scripting or tooling in your package has issues running in Node@12+ due to this inclusion, excluding it will ease the local requirements on import statements in your code.
```

> tooling the references your package in leveraging it more efficiently

not sure i totally grok what you mean with this, its a bit confusing

Reply:
Still having trouble with this one, new version coming to the preview soon.